foreshadow/foreshadow.py

```diff
@@ -555,3 +555,42 @@ def configure_multiprocessing(self, n_job: int = 1) -> NoReturn:
 
         """
         self.X_preparer.cache_manager["config"][ConfigKey.N_JOBS] = n_job
+
+    def to_pickle(self, path: str) -> NoReturn:
+        """Pickle the foreshadow object with the best pipeline estimator.
+
+        Args:
+            path: the pickle file path
+
+        """
+        import pickle
+
+        if (
+            isinstance(self.estimator.estimator, AutoEstimator)
+            and self.estimator.estimator.estimator.fitted_pipeline_ is not None
+        ):
+            self.estimator = (
+                self.estimator.estimator.estimator.fitted_pipeline_
+            )
+            # updating the estimator above will not update the reference in
+            # the pipeline instance as it still points to the old object.
+            self.pipeline.steps[1] = ("estimator", self.estimator)
+        with open(path, "wb") as fopen:
+            pickle.dump(self, fopen)
+
+    @classmethod
+    def from_pickle(cls, path: str):
```

Comment:
It seems we only need to support the pickling of the fitted pipeline, which should be a sklearn pipeline object. I suggest we make it more explicit and only do the pickling when Foreshadow is already fitted. If we do that, we don't need the from_pickle function any more, User can simply use the built-in pickle object to unpickle a pipeline object.

Reply:
Done.